src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStarImportCheck.java

```diff
@@ -220,8 +220,7 @@ public void visitToken(final DetailAST ast) {
             final DetailAST startingDot = ast.getFirstChild();
             logsStarredImportViolation(startingDot);
         }
-        else if (!allowStaticMemberImports
-            && ast.getType() == TokenTypes.STATIC_IMPORT) {
+        else if (!allowStaticMemberImports) {
```

Comment:
@wilcoln Is there some other way to change the code to make more sense and work with an invalid AST?

These 2 if conditions are very similar except for these lines from each:
> final DetailAST startingDot = ast.getFirstChild();
> final DetailAST startingDot = ast.getFirstChild().getNextSibling();

My current idea is the first if statement to change it to:
````
if (ast.getType() == TokenTypes.IMPORT) {
...
} else if (!allowStaticMemberImports) {
...
}
````

You agree that there is no reason for imports to go to the else if as it will never be a valid reason?

Reply:
I agree. It doesn't make much sense indeed. It's like we're relying on some bother effect to keep things straight.

Reply:
I just ran ` ./.ci/pitest.sh pitest-imports
` with this new code 
```
public void visitToken(final DetailAST ast) {
        if(ast.getType() == TokenTypes.IMPORT){
            if (!allowClassImports) {
                final DetailAST startingDot = ast.getFirstChild();
                logsStarredImportViolation(startingDot);
            }
        }
        else if (!allowStaticMemberImports) {
            // must navigate past the static keyword
            final DetailAST startingDot = ast.getFirstChild().getNextSibling();
            logsStarredImportViolation(startingDot);
        }
    }
```
and it works just fine. Should I make a PR?